### PR TITLE
WIP: New key class

### LINKED ISSUE #244: New class for list of keys
for better printing in tibbles. Required after #239.

### PR DESCRIPTION
**Pull request by TSchiefer**
The description here is outdated, method `pillar_shaft.dm_keys()` solved it:

> the column-header looks already alright, but the elements themselves won't display correctly.

> When trying to find it, I realized, that in `pillar::colonnade()` in line 36:
> ```r
>  proxy <- structure(x, has_row_id = has_row_id)
> ```
> it looks as I would think it should, but after the following line 37:
> ```r
>  ret <- structure(proxy, class = "pillar_colonnade")
> ```
> it displays it in the list view that we want to avoid. It would be nice if you could have a look at it.

still there are a few minor issues

fixes #244

### CODE REVIEW

**Comment by TSchiefer** [comment]
Update: now it renders correctly I think. The method `pillar::pillar_shaft.dm_keys()` needed to be implemented

File: R/dm.R [hunk 1]
```diff
@@ -568,6 +568,27 @@ format.zoomed_df <- function(x, ..., n = NULL, width = NULL, n_extra = NULL) {
   NextMethod()
 }
 
+print.dm_keys <- function(x, ..., n = 3, width = NULL, n_extra = NULL) {
```
Comment (krlmlr) [inline R/dm.R]: I think this doesn't have an effect. Can we remove it without breaking functionality?

File: R/dm.R [hunk 2]
```diff
@@ -568,6 +568,27 @@ format.zoomed_df <- function(x, ..., n = NULL, width = NULL, n_extra = NULL) {
...
+  format(x, ..., n = n, width = NULL, n_extra = NULL)
+}
+
+#' @export
+format.dm_keys <- function(x, ..., n = NULL, width = NULL, n_extra = NULL) {
+  print(map_chr(x, commas, n))
```
Comment (krlmlr) [inline R/dm.R]: `format()` needs to return a character vector.

File: R/dm.R [hunk 3]
```diff
@@ -568,6 +568,27 @@ format.zoomed_df <- function(x, ..., n = NULL, width = NULL, n_extra = NULL) {
...
+  invisible(x)
+}
+
+vec_ptype_abbr.dm_keys <- function(x) {
+  "keys"
+}
+
+pillar_shaft.dm_keys <- function(x, n = 3) {
```
Comment (krlmlr) [inline R/dm.R]:
```suggestion
#' @export
pillar_shaft.dm_keys <- function(x, n = 3) {
```

**Comment by TSchiefer** [comment]
Sorry, I didn't find the time for this today. (My) tomorrow morning, if it's early enough for you.

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**